compiler/fir/tree/src/org/jetbrains/kotlin/fir/FirRenderer.kt

```diff
@@ -972,7 +972,11 @@ class FirRenderer(builder: StringBuilder, private val mode: RenderMode = RenderM
     }
 
     override fun visitFunctionTypeRef(functionTypeRef: FirFunctionTypeRef) {
+        functionTypeRef.annotations.dropExtensionFunctionAnnotation().renderAnnotations()
```

Comment:
Should we check this flag, `RenderMode.renderAnnotation`? https://github.com/JetBrains/kotlin/blob/master/compiler/fir/tree/src/org/jetbrains/kotlin/fir/FirRenderer.kt#L59

Reply:
[It's already read in the `renderAnnotations()` method](https://github.com/JetBrains/kotlin/blob/6cbdcbe53b321da6809b00a15bb98bce673bfe26/compiler/fir/tree/src/org/jetbrains/kotlin/fir/FirRenderer.kt#L195)